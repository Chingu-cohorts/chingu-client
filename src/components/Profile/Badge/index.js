import React from "react";

const Badge = props => <div className="chip">{props.name}</div>;

export default Badge;
