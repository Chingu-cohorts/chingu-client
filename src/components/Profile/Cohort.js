import React from "react";

import "./Cohort.css";

const Cohort = props => (
  <div className="cohort">
    <h4>{props.name}</h4>
    <p>
      {props.members} Members
    </p>
  </div>
);

export default Cohort;
