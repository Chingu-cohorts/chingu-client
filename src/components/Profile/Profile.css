.header {
  margin-top: 5rem;
  padding: 1rem 1rem 0 1rem;
}

.header img {
  width: 100%;
  border-radius: 50%;
  position: relative;
  top: -2rem;
}

.header p {
  margin-top: 0;
}

.header h2 {
  margin: 0;
}

.header ul li {
  display: inline-block;
  margin-right: 1rem;
}
