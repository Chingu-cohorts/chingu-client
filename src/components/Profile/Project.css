.project {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.project .project-container {
  margin-top: 1rem;
}

.project .project-container .row {
  margin-bottom: 0;
}

.project img {
  max-width: 100%;
  max-height: 100%;
  padding: 0.5rem;
}

.project ul li {
  display: inline-block;
  margin-right: 1rem;
}

.project strong {
  font-weight: 600;
}
