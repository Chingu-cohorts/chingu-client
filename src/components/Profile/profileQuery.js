import gql from "graphql-tag";

const userQuery = gql`
  query userProfile($userId: ID!) {
    user(user_id: $userId) {
      first_name
      last_name
      bio
      github_url
      twitter_url
      linkedin_url
      blog_url
      portfolio_url
      projects {
        title
        description
        github_url
        project_url
        users {
          username
        }
      }
      cohorts {
        users {
          username
        }
      }
    }
  }
`;

export default userQuery;
