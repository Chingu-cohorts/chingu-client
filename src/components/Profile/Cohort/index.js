import React from "react";

import "./Cohort.css";

const Cohort = props => (
  <div className="col s12 m12 grey lighten-4 cohort">
    <h5>{props.name}</h5>
    <p className="grey-text lighten-1">
      {props.members} Members, {props.countries} Countries, {props.projects}{" "}
      Projects
    </p>
  </div>
);

export default Cohort;
