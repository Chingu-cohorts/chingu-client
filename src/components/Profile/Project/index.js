import React from "react";
import { Card } from "antd";

import "./Project.css";

const Project = props => (
  <Card>
    <div className="image">
      <img
        alt="example"
        width="100%"
        src="https://os.alipayobjects.com/rmsportal/QBnOOoLaAfKPirc.png"
      />
    </div>
    <div className="content">
      <h3>{props.title}</h3>
      <p>{props.description}</p>
      <p>
        <ul>
          <li>
            {props.github_url ? (
              <a href={props.github_url}>Github</a>
            ) : (
              "No GitHub"
            )}
          </li>
          <li>
            {props.project_url ? (
              <a href={props.project_url}>Live</a>
            ) : (
              "No Project URL"
            )}
          </li>
        </ul>
      </p>
      <p>
        <strong>Team</strong>
        <ul>
          {props.users.map(user => {
            return <li>{user.username}</li>;
          })}
        </ul>
      </p>
    </div>
  </Card>
);

export default Project;
