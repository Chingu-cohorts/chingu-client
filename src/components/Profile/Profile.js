import React, { Component } from "react";
import { graphql } from "react-apollo";
import { Row, Col, Layout, Spin } from "antd";

import Project from "./Project";
import Cohort from "./Cohort";
import profileQuery from "./profileQuery";
import "./Profile.css";

const { Content } = Layout;

export class Profile extends Component {
  render() {
    let { loading, error, user } = this.props.data;
    if (loading) {
      return (
        <Content>
          <div className="text-center">
            <Spin />
          </div>
        </Content>
      );
    } else if (error) {
      return (
        <Content>
          <h1>An error ocurred</h1>
        </Content>
      );
    } else if (!user) {
      return (
        <Content>
          <h1>User not found</h1>
        </Content>
      );
    }

    return (
      <Content>
        <div className="section header" style={{ marginTop: "3rem" }}>
          <Row type="flex" justify="center">
            <Col sm={24} md={8} lg={6}>
              <img
                src={`https://api.adorable.io/avatars/250/${user.email}.png`}
                alt="Avatar"
              />
              <p className="text-center">
                {user.country ? user.country.name : "No location provided"}
              </p>
            </Col>
            <Col sm={24} md={16} lg={18}>
              <h2>
                {user.first_name} {user.last_name}
              </h2>
              <ul>
                {user.github_url ? (
                  <li>
                    <a href={user.github_url}>GitHub</a>
                  </li>
                ) : null}

                {user.twitter_url ? (
                  <li>
                    <a href={user.twitter_url}>Twitter</a>
                  </li>
                ) : null}

                {user.blog_url ? (
                  <li>
                    <a href={user.blog_url}>Blog</a>
                  </li>
                ) : null}
              </ul>
              <p className="bio">
                {user.bio ? user.bio : "User does not have a bio yet."}
              </p>
            </Col>
          </Row>
        </div>

        <div className="section profile-content">
          <Row type="flex" justify="center" gutter={16}>
            <Col sm={24} md={16}>
              <h3>Projects</h3>
              <Row type="flex" gutter={16}>
                {user.projects.length > 0
                  ? user.projects.map((project, index) => {
                      return (
                        <Col md={12}>
                          <Project {...project} key={index} />
                        </Col>
                      );
                    })
                  : "User doesn't have any projects."}
              </Row>
            </Col>
            <Col sm={24} md={8}>
              <h3>Cohorts</h3>
              <Row type="flex" gutter={16}>
                {user.cohorts.length > 0
                  ? user.cohorts.map((cohort, index) => {
                      return <Cohort {...cohort} />;
                    })
                  : "User hasn't been part of any cohorts."}
              </Row>
            </Col>
          </Row>
        </div>
      </Content>
    );
  }
}

export default graphql(profileQuery, {
  options: ownProps => ({
    variables: {
      userId: ownProps.match.params.id
    }
  })
})(Profile);
