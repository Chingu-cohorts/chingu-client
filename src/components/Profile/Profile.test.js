import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";

import { Profile } from "./Profile";

describe("Profile component", () => {
  it("renders without crashing", () => {
    const props = {
      match: {
        params: {
          id: 1
        }
      }
    };
    const data = {
      user: {
        first_name: "Andrés",
        last_name: "Pérez",
        github_url: "https://github.com",
        projects: [],
        cohorts: []
      }
    };

    const wrapper = shallow(<Profile {...props} data={data} />);

    expect(wrapper).toMatchSnapshot();
  });
});
