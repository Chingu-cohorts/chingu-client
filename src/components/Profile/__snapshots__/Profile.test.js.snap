// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    data={
        Object {
            "user": Object {
              "cohorts": Array [],
              "first_name": "Andrés",
              "github_url": "https://github.com",
              "last_name": "Pérez",
              "projects": Array [],
            },
          }
    }
    match={
        Object {
            "params": Object {
              "id": 1,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="section header"
          style={
                    Object {
                              "marginTop": "3rem",
                            }
          }
>
          <Row
                    gutter={0}
                    justify="center"
                    type="flex"
          >
                    <Col
                              lg={6}
                              md={8}
                              sm={24}
                    >
                              <img
                                        alt="Avatar"
                                        src="https://api.adorable.io/avatars/250/undefined.png"
                              />
                              <p
                                        className="text-center"
                              >
                                        No location provided
                              </p>
                    </Col>
                    <Col
                              lg={18}
                              md={16}
                              sm={24}
                    >
                              <h2>
                                        Andrés
                                         
                                        Pérez
                              </h2>
                              <ul>
                                        <li>
                                                  <a
                                                            href="https://github.com"
                                                  >
                                                            GitHub
                                                  </a>
                                        </li>
                              </ul>
                              <p
                                        className="bio"
                              >
                                        User does not have a bio yet.
                              </p>
                    </Col>
          </Row>
</div>,
        <div
          className="section profile-content"
>
          <Row
                    gutter={16}
                    justify="center"
                    type="flex"
          >
                    <Col
                              md={16}
                              sm={24}
                    >
                              <h3>
                                        Projects
                              </h3>
                              <Row
                                        gutter={16}
                                        type="flex"
                              >
                                        User doesn't have any projects.
                              </Row>
                    </Col>
                    <Col
                              md={8}
                              sm={24}
                    >
                              <h3>
                                        Cohorts
                              </h3>
                              <Row
                                        gutter={16}
                                        type="flex"
                              >
                                        User hasn't been part of any cohorts.
                              </Row>
                    </Col>
          </Row>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Row
            gutter={0}
            justify="center"
            type="flex"
>
            <Col
                        lg={6}
                        md={8}
                        sm={24}
            >
                        <img
                                    alt="Avatar"
                                    src="https://api.adorable.io/avatars/250/undefined.png"
                        />
                        <p
                                    className="text-center"
                        >
                                    No location provided
                        </p>
            </Col>
            <Col
                        lg={18}
                        md={16}
                        sm={24}
            >
                        <h2>
                                    Andrés
                                     
                                    Pérez
                        </h2>
                        <ul>
                                    <li>
                                                <a
                                                            href="https://github.com"
                                                >
                                                            GitHub
                                                </a>
                                    </li>
                        </ul>
                        <p
                                    className="bio"
                        >
                                    User does not have a bio yet.
                        </p>
            </Col>
</Row>,
          "className": "section header",
          "style": Object {
            "marginTop": "3rem",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Col
                lg={6}
                md={8}
                sm={24}
>
                <img
                                alt="Avatar"
                                src="https://api.adorable.io/avatars/250/undefined.png"
                />
                <p
                                className="text-center"
                >
                                No location provided
                </p>
</Col>,
              <Col
                lg={18}
                md={16}
                sm={24}
>
                <h2>
                                Andrés
                                 
                                Pérez
                </h2>
                <ul>
                                <li>
                                                <a
                                                                href="https://github.com"
                                                >
                                                                GitHub
                                                </a>
                                </li>
                </ul>
                <p
                                className="bio"
                >
                                User does not have a bio yet.
                </p>
</Col>,
            ],
            "gutter": 0,
            "justify": "center",
            "type": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt="Avatar"
                    src="https://api.adorable.io/avatars/250/undefined.png"
/>,
                  <p
                    className="text-center"
>
                    No location provided
</p>,
                ],
                "lg": 6,
                "md": 8,
                "sm": 24,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Avatar",
                    "src": "https://api.adorable.io/avatars/250/undefined.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No location provided",
                    "className": "text-center",
                  },
                  "ref": null,
                  "rendered": "No location provided",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h2>
                    Andrés
                     
                    Pérez
</h2>,
                  <ul>
                    <li>
                                        <a
                                                            href="https://github.com"
                                        >
                                                            GitHub
                                        </a>
                    </li>
</ul>,
                  <p
                    className="bio"
>
                    User does not have a bio yet.
</p>,
                ],
                "lg": 18,
                "md": 16,
                "sm": 24,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Andrés",
                      " ",
                      "Pérez",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Andrés",
                    " ",
                    "Pérez",
                  ],
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li>
                        <a
                                                href="https://github.com"
                        >
                                                GitHub
                        </a>
</li>,
                      null,
                      null,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href="https://github.com"
>
                          GitHub
</a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "GitHub",
                          "href": "https://github.com",
                        },
                        "ref": null,
                        "rendered": "GitHub",
                        "type": "a",
                      },
                      "type": "li",
                    },
                    null,
                    null,
                  ],
                  "type": "ul",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "User does not have a bio yet.",
                    "className": "bio",
                  },
                  "ref": null,
                  "rendered": "User does not have a bio yet.",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Row
            gutter={16}
            justify="center"
            type="flex"
>
            <Col
                        md={16}
                        sm={24}
            >
                        <h3>
                                    Projects
                        </h3>
                        <Row
                                    gutter={16}
                                    type="flex"
                        >
                                    User doesn't have any projects.
                        </Row>
            </Col>
            <Col
                        md={8}
                        sm={24}
            >
                        <h3>
                                    Cohorts
                        </h3>
                        <Row
                                    gutter={16}
                                    type="flex"
                        >
                                    User hasn't been part of any cohorts.
                        </Row>
            </Col>
</Row>,
          "className": "section profile-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Col
                md={16}
                sm={24}
>
                <h3>
                                Projects
                </h3>
                <Row
                                gutter={16}
                                type="flex"
                >
                                User doesn't have any projects.
                </Row>
</Col>,
              <Col
                md={8}
                sm={24}
>
                <h3>
                                Cohorts
                </h3>
                <Row
                                gutter={16}
                                type="flex"
                >
                                User hasn't been part of any cohorts.
                </Row>
</Col>,
            ],
            "gutter": 16,
            "justify": "center",
            "type": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h3>
                    Projects
</h3>,
                  <Row
                    gutter={16}
                    type="flex"
>
                    User doesn't have any projects.
</Row>,
                ],
                "md": 16,
                "sm": 24,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Projects",
                  },
                  "ref": null,
                  "rendered": "Projects",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "User doesn't have any projects.",
                    "gutter": 16,
                    "type": "flex",
                  },
                  "ref": null,
                  "rendered": "User doesn't have any projects.",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h3>
                    Cohorts
</h3>,
                  <Row
                    gutter={16}
                    type="flex"
>
                    User hasn't been part of any cohorts.
</Row>,
                ],
                "md": 8,
                "sm": 24,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cohorts",
                  },
                  "ref": null,
                  "rendered": "Cohorts",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "User hasn't been part of any cohorts.",
                    "gutter": 16,
                    "type": "flex",
                  },
                  "ref": null,
                  "rendered": "User hasn't been part of any cohorts.",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="section header"
            style={
                        Object {
                                    "marginTop": "3rem",
                                  }
            }
>
            <Row
                        gutter={0}
                        justify="center"
                        type="flex"
            >
                        <Col
                                    lg={6}
                                    md={8}
                                    sm={24}
                        >
                                    <img
                                                alt="Avatar"
                                                src="https://api.adorable.io/avatars/250/undefined.png"
                                    />
                                    <p
                                                className="text-center"
                                    >
                                                No location provided
                                    </p>
                        </Col>
                        <Col
                                    lg={18}
                                    md={16}
                                    sm={24}
                        >
                                    <h2>
                                                Andrés
                                                 
                                                Pérez
                                    </h2>
                                    <ul>
                                                <li>
                                                            <a
                                                                        href="https://github.com"
                                                            >
                                                                        GitHub
                                                            </a>
                                                </li>
                                    </ul>
                                    <p
                                                className="bio"
                                    >
                                                User does not have a bio yet.
                                    </p>
                        </Col>
            </Row>
</div>,
          <div
            className="section profile-content"
>
            <Row
                        gutter={16}
                        justify="center"
                        type="flex"
            >
                        <Col
                                    md={16}
                                    sm={24}
                        >
                                    <h3>
                                                Projects
                                    </h3>
                                    <Row
                                                gutter={16}
                                                type="flex"
                                    >
                                                User doesn't have any projects.
                                    </Row>
                        </Col>
                        <Col
                                    md={8}
                                    sm={24}
                        >
                                    <h3>
                                                Cohorts
                                    </h3>
                                    <Row
                                                gutter={16}
                                                type="flex"
                                    >
                                                User hasn't been part of any cohorts.
                                    </Row>
                        </Col>
            </Row>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Row
              gutter={0}
              justify="center"
              type="flex"
>
              <Col
                            lg={6}
                            md={8}
                            sm={24}
              >
                            <img
                                          alt="Avatar"
                                          src="https://api.adorable.io/avatars/250/undefined.png"
                            />
                            <p
                                          className="text-center"
                            >
                                          No location provided
                            </p>
              </Col>
              <Col
                            lg={18}
                            md={16}
                            sm={24}
              >
                            <h2>
                                          Andrés
                                           
                                          Pérez
                            </h2>
                            <ul>
                                          <li>
                                                        <a
                                                                      href="https://github.com"
                                                        >
                                                                      GitHub
                                                        </a>
                                          </li>
                            </ul>
                            <p
                                          className="bio"
                            >
                                          User does not have a bio yet.
                            </p>
              </Col>
</Row>,
            "className": "section header",
            "style": Object {
              "marginTop": "3rem",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Col
                  lg={6}
                  md={8}
                  sm={24}
>
                  <img
                                    alt="Avatar"
                                    src="https://api.adorable.io/avatars/250/undefined.png"
                  />
                  <p
                                    className="text-center"
                  >
                                    No location provided
                  </p>
</Col>,
                <Col
                  lg={18}
                  md={16}
                  sm={24}
>
                  <h2>
                                    Andrés
                                     
                                    Pérez
                  </h2>
                  <ul>
                                    <li>
                                                      <a
                                                                        href="https://github.com"
                                                      >
                                                                        GitHub
                                                      </a>
                                    </li>
                  </ul>
                  <p
                                    className="bio"
                  >
                                    User does not have a bio yet.
                  </p>
</Col>,
              ],
              "gutter": 0,
              "justify": "center",
              "type": "flex",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <img
                      alt="Avatar"
                      src="https://api.adorable.io/avatars/250/undefined.png"
/>,
                    <p
                      className="text-center"
>
                      No location provided
</p>,
                  ],
                  "lg": 6,
                  "md": 8,
                  "sm": 24,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Avatar",
                      "src": "https://api.adorable.io/avatars/250/undefined.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "No location provided",
                      "className": "text-center",
                    },
                    "ref": null,
                    "rendered": "No location provided",
                    "type": "p",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <h2>
                      Andrés
                       
                      Pérez
</h2>,
                    <ul>
                      <li>
                                            <a
                                                                  href="https://github.com"
                                            >
                                                                  GitHub
                                            </a>
                      </li>
</ul>,
                    <p
                      className="bio"
>
                      User does not have a bio yet.
</p>,
                  ],
                  "lg": 18,
                  "md": 16,
                  "sm": 24,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Andrés",
                        " ",
                        "Pérez",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Andrés",
                      " ",
                      "Pérez",
                    ],
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <li>
                          <a
                                                    href="https://github.com"
                          >
                                                    GitHub
                          </a>
</li>,
                        null,
                        null,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <a
                            href="https://github.com"
>
                            GitHub
</a>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "GitHub",
                            "href": "https://github.com",
                          },
                          "ref": null,
                          "rendered": "GitHub",
                          "type": "a",
                        },
                        "type": "li",
                      },
                      null,
                      null,
                    ],
                    "type": "ul",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "User does not have a bio yet.",
                      "className": "bio",
                    },
                    "ref": null,
                    "rendered": "User does not have a bio yet.",
                    "type": "p",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Row
              gutter={16}
              justify="center"
              type="flex"
>
              <Col
                            md={16}
                            sm={24}
              >
                            <h3>
                                          Projects
                            </h3>
                            <Row
                                          gutter={16}
                                          type="flex"
                            >
                                          User doesn't have any projects.
                            </Row>
              </Col>
              <Col
                            md={8}
                            sm={24}
              >
                            <h3>
                                          Cohorts
                            </h3>
                            <Row
                                          gutter={16}
                                          type="flex"
                            >
                                          User hasn't been part of any cohorts.
                            </Row>
              </Col>
</Row>,
            "className": "section profile-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Col
                  md={16}
                  sm={24}
>
                  <h3>
                                    Projects
                  </h3>
                  <Row
                                    gutter={16}
                                    type="flex"
                  >
                                    User doesn't have any projects.
                  </Row>
</Col>,
                <Col
                  md={8}
                  sm={24}
>
                  <h3>
                                    Cohorts
                  </h3>
                  <Row
                                    gutter={16}
                                    type="flex"
                  >
                                    User hasn't been part of any cohorts.
                  </Row>
</Col>,
              ],
              "gutter": 16,
              "justify": "center",
              "type": "flex",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <h3>
                      Projects
</h3>,
                    <Row
                      gutter={16}
                      type="flex"
>
                      User doesn't have any projects.
</Row>,
                  ],
                  "md": 16,
                  "sm": 24,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Projects",
                    },
                    "ref": null,
                    "rendered": "Projects",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "User doesn't have any projects.",
                      "gutter": 16,
                      "type": "flex",
                    },
                    "ref": null,
                    "rendered": "User doesn't have any projects.",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <h3>
                      Cohorts
</h3>,
                    <Row
                      gutter={16}
                      type="flex"
>
                      User hasn't been part of any cohorts.
</Row>,
                  ],
                  "md": 8,
                  "sm": 24,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Cohorts",
                    },
                    "ref": null,
                    "rendered": "Cohorts",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "User hasn't been part of any cohorts.",
                      "gutter": 16,
                      "type": "flex",
                    },
                    "ref": null,
                    "rendered": "User hasn't been part of any cohorts.",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
